packages/sanity/src/router/utils/base64url.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * `atob()` and `btoa()` do not support Unicode characters outside of the Latin1 range,
+ * but we obviously want to support the full range of Unicode characters in our router.
+ *
+ * Additionally, we would prefer not to use characters like `+` and `=` in URLs, as they
+ * have specific meanings there and may be misinterpreted. Thus, this uses base64url instead
+ * of the more common base64.
+ */
+
+/**
+ * Encodes a string as base64url
+ *
+ * @param str - String to encode
+ * @returns Encoded string
+ * @internal
+ */
+export function encodeBase64Url(str: string): string {
+  return encodeBase64(str).replace(/\//g, '_').replace(/\+/g, '-').replace(/[=]+$/, '')
+}
+
+/**
+ * Decodes a base64url-encoded string
+ *
+ * @param str - String to decode
+ * @returns Decoded string
+ * @internal
+ */
+export function decodeBase64Url(str: string): string {
```

Comment:
the encode removes `=` chars, but the decode does not add them back. I'm assuming that is intentional? Should that be explicitly documented?

Reply:
Shouldn't need to? [RFC 4648 § 5 ](https://datatracker.ietf.org/doc/html/rfc4648#section-5) states:
> The pad character "=" is typically percent-encoded when used in an URI [[9](https://datatracker.ietf.org/doc/html/rfc4648#ref-9)], but if the data length is known implicitly, this can be avoided by skipping the padding; see [section 3.2](https://datatracker.ietf.org/doc/html/rfc4648#section-3.2).

As long as we always attempt to decode using base64url _prior_ to attempting base64, this should work fine